test/run-spec-wasm2c.py

```diff
@@ -243,9 +253,15 @@ def _WriteCommand(self, command):
     def _WriteModuleCommand(self, command):
         self.module_idx += 1
         self.out_file.write('%s_init_module();\n' % self.GetModulePrefix())
-        self.out_file.write('%s_instance_t %s;\n' % (self.GetModulePrefix(), self.GetModuleInstanceName()))
         self._WriteModuleInitCall(command, False)
 
+    def _WriteModuleInstances(self):
+        idx = 0
+        for command in self.commands:
+            if IsModuleCommand(command):
+                self.out_file.write('%s_instance_t %s;\n' % (self.GetModulePrefix(idx), self.GetModuleInstanceName(idx)))
+                idx += 1
```

Comment:
Same here, is this related to the splitting up?

Reply:
Yes, this is most of the first commit. We're declaring all of the instance_t structs upfront (as globals) so that any of the split-up run_spec_tests_**N** functions can refer to any of them.